Packages/com.unity.inputsystem/InputSystem/API/Input.actions.cs

```diff
@@ -0,0 +1,534 @@
+            }
+            else
+            {
+                var runtimeAssetLoader = (IGlobalActionsRuntimeAsset)Activator.CreateInstance(runtimeAssetLoaderType);
+                InputSystem.actions = runtimeAssetLoader.Load();
+            }
+#endif
+            if (InputSystem.actions == null)
+            {
+                Debug.LogError($"Couldn't initialize global input actions");
+                return;
+            }
+
+            // TODO: Once the source generator is running, only the actions that have actually been used should be enabled
+            InputSystem.actions.Enable();
+        }
+
+        internal static void ShutdownGlobalActions()
+        {
+            if (InputSystem.actions == null)
+                return;
+
+            InputSystem.actions.Disable();
+            InputSystem.actions = null;
+        }
+    }
+
+    public interface IGlobalActionsRuntimeAsset
+    {
+        InputActionAsset Load();
+    }
+
+    /// <summary>
+    /// A strongly-typed wrapper for an Input Action.
+    /// </summary>
+    /// <typeparam name="TActionType">The type that will be used in calls to <see cref="InputAction.ReadValue{T}"/></typeparam>
+    public class Input<TActionType> where TActionType : struct
+    {
+        /// <summary>
+        /// Enables access to the underlying Input Action for advanced functionality such as rebinding.
+        /// </summary>
+        public InputAction action => m_Action;
+
+
+        /// <see cref="InputAction.IsPressed"/>
+        public bool isPressed => m_Action.IsPressed();
+        /// <see cref="InputAction.WasPressedThisFrame"/>
+        public bool wasPressedThisFrame => m_Action.WasPressedThisFrame();
+        /// <see cref="InputAction.WasReleasedThisFrame"/>
+        public bool wasReleasedThisFrame => m_Action.WasReleasedThisFrame();
+
+        /// <summary>
+        /// Returns the current value of the Input Action.
+        /// </summary>
+        public TActionType value
+        {
+            get
+            {
+                try
+                {
+                    // it should be unusual for ReadValue to throw because in most cases instances of this class
+                    // will be created through the source generator, and that can catch mismatched control types
+                    // at compile time and throw compiler errors, but it will always be possible to dynamically
+                    // add incompatible bindings, and the best we can do then is to catch the exceptions
+                    // thrown when we try to read from those controls.
+                    return m_Action.ReadValue<TActionType>();
+                }
+                catch (InvalidOperationException ex)
+                {
+                    Debug.LogWarning(ex.Message);
+                }
+
+                return m_Action.ReadDefaultValue<TActionType>(
+                    new BindingIndex(m_Action, m_Action.activeBindingIndex, BindingIndex.IndexType.IncludeCompositeParts)
```

Comment:
Is this creating any Garbage?
If so perhaps we could create a bindingIndex up front and update the value with the active binding index if needed/changes.

Not critical for 0.8.0-pre